common/models/user.js

```diff
@@ -360,20 +367,19 @@ module.exports = function(User) {
     // Email model
     var Email = options.mailer || this.constructor.email || loopback.getModelByType(loopback.Email);
 
-    crypto.randomBytes(64, function(err, buf) {
-      if (err) {
-        fn(err);
-      } else {
-        user.verificationToken = buf.toString('hex');
-        user.save(function(err) {
-          if (err) {
-            fn(err);
-          } else {
-            sendEmail(user);
-          }
-        });
-      }
-    });
+    // Set a default token generation function if one is nto provided
+    var tokenGenerator = options.generateVerificationToken || User.generateVerificationToken;
+
+    tokenGenerator.apply(user, [function(token) {
```

Comment:
Okay, I'm fine with it being an argument. I guess my thought was to be consistent with the `verify()` method, which uses it as context versus argument. But again, I'm totally open. I'll wait until later today to make the change in case there are other opinions. :)